api/v1alpha1/selfnoderemediation_types.go

```diff
@@ -38,12 +39,13 @@ type RemediationStrategyType string
 // SelfNodeRemediationSpec defines the desired state of SelfNodeRemediation
 type SelfNodeRemediationSpec struct {
 	//RemediationStrategy is the remediation method for unhealthy nodes.
-	//Currently, it could be either "ResourceDeletion" or "OutOfServiceTaint".
-	//The first will iterate over all pods and VolumeAttachment related to the unhealthy node and delete them.
-	//The latter will add the out-of-service taint which is a new well-known taint "node.kubernetes.io/out-of-service"
+	//Currently, it could be either "Automatic", "OutOfServiceTaint" or "ResourceDeletion".
+	//ResourceDeletion will iterate over all pods and VolumeAttachment related to the unhealthy node and delete them.
+	//OutOfServiceTaint will add the out-of-service taint which is a new well-known taint "node.kubernetes.io/out-of-service"
 	//that enables automatic deletion of pv-attached pods on failed nodes, "OutOfServiceTaint" is only supported on clusters with k8s version 1.26+ or OCP/OKD version 4.13+.
-	// +kubebuilder:default:="ResourceDeletion"
-	// +kubebuilder:validation:Enum=ResourceDeletion;OutOfServiceTaint
+	// Automatic will choose the most appropriate strategy during runtime.
```

Comment:
This is kind of vague. Is that on purpose or could we just write which remediation will be used most of the time?

``` suggestion
Automatic will choose the most appropriate strategy during runtime. By default, OutOfServiceTaint will be selected If the running Kubernetes can support the new taint, otherwise, ResourceDeletion will be selected.
```